messages/jsonschema/scripts/templates/ruby.rb.erb

```diff
@@ -24,6 +24,29 @@ module Cucumber
       attr_reader :<%= underscore(property_name) %>
     <%- end -%>
 
+      def self.from_json(json)
+        from_camel_symbol_hash(JSON.parse(json, {symbolize_names: true}))
+      end
+
+      def self.from_camel_symbol_hash(hash)
+        return nil if hash.nil?
+        self.new(
+      <%-
+        schema['properties'].each do |property_name, property|
+          ref = property['$ref']
+          items_ref = property.dig('items', '$ref')
+      -%>
+        <%- if items_ref -%>
```

Comment:
I can't easily comprehend what each leg of the logic is doing here. Can we extract some methods?

Reply:
The first one is for arrays of objects
The second one is for objects
The third is for primitives and arrays of primitives.

I agree we should extract a method up into the `Ruby` class